traceon/geometry.py

```diff
@@ -1134,7 +1131,180 @@ def rectangle_xy(xmin, xmax, ymin, ymax):
     @staticmethod
     def aperture(height, radius, extent, z=0.):
         return Path.aperture(height, radius, extent, z=z).revolve_z()
-     
+    
+    def get_boundary_paths(self):
+        """Get the boundary paths of the surface.
+        Computes the boundary paths (edges) of the surface and combines them into a `PathCollection`.
+        Non-closed paths get filtered out when closed paths are present, as only closed paths 
+        represent true boundaries in this case.
+        Note that this function might behave unexpectedly for surfaces without any boundaries (e.g a sphere).
+
+        Returns
+        ----------------------------
+        PathCollection representing the boundary paths of the surface"""
+        
+        b1 = Path(lambda u: self(u, 0.), self.path_length1, self.breakpoints1, self.name)
+        b2 = Path(lambda u: self(u, self.path_length2), self.path_length1, self.breakpoints1, self.name)
+        b3 = Path(lambda v: self(0., v), self.path_length2, self.breakpoints2, self.name)
+        b4 = Path(lambda v: self(self.path_length1, v), self.path_length2, self.breakpoints2, self.name)
+        
+        boundary = b1 + b2 + b3 + b4
+
+        if any([b.is_closed() for b in boundary.paths]):
+            boundary = PathCollection([b for b in boundary.paths if b.is_closed()])
+        
+        return boundary
+
+    def extrude_boundary(self, vector, add_cap=True):
+        """
+        Extrude the boundary paths of the surface along a vector. The vector gives both
+        the length and the direction of the extrusion.
+
+        Parameters
+        -------------------------
+        vector: (3,) float
+            The direction and length (norm of the vector) to extrude by.
+        add_cap: bool
+            Whether to cap the extrusion by adding a copy of the original surface 
+            moved by the extrusion vector to the resulting SurfaceCollection.
+
+        Returns
+        -------------------------
+        SurfaceCollection"""
+
+        boundary = self.get_boundary_paths()
+        extruded_boundary_paths = []
+
+        for b in boundary.paths:
+            try: extruded_boundary_paths.append(b.extrude(vector))
+            except AssertionError: pass
```

Comment:
Please don't use exceptions for normal control flow, as this make the program hard to read. Why does the exception trigger? If because of zero path length consider the code below
```Python3
extruded_boundary_paths = [b.extrude(vector) for b in boundary_paths if b.path_length > 0]
```

Reply:
The exception triggers in case one of the boundary paths results in a surface with zero area, for instance if one of the boundaries is a line segment along the z-axis and we try to revolve around this axis. I don't immediately see a quick check for this. I could try to implement a function that checks if a straight line along an axis (e.g. compute the tangent vector along the path and check if its constant) which might be useful in a broader sense as well. Do you have any other suggestions?

Reply:
fixed by removing the assertion error altogether. I have not observed problems with the mesher after this, but this might need some more investigation.